src/ui/PromotedResultsBadge/PromotedResultsBadge.ts

```diff
@@ -0,0 +1,187 @@
+import { Component } from '../../ui/Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, Initialization, l, $$, ResultListEvents, Dom } from '../../Core';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IDisplayedNewResultEventArgs } from '../../events/ResultListEvents';
+import 'styling/_PromotedResultsBadge';
```

Comment:
Compiles it with sass when imported, in dev mode it's extracted in an "inline" style tag (so it can recompile/reload on change), in production it is extracted in a single file.

Compared to the interface editor, it is not adding hash/token to class names to make sure it is "encapsulated". It would be very bad if we did that, since people would not be able to customize it with external CSS rules.